io-sim-classes/src/Control/Monad/Class/MonadAsync.hs

```diff
@@ -41,7 +45,10 @@ import           Data.Foldable (fold)
 import           Data.Kind (Type)
 import           Data.Proxy
 
-class (Functor async, MonadSTMTx stm) => MonadAsyncSTM async stm where
+class ( Functor async
+      , MonadSTMTx stm tvar tmvar tqueue tbqueue
+      )
+      => MonadAsyncSTM async stm tvar tmvar tqueue tbqueue where
```

Comment:
@coot Would you add a comment explaining the reason for MPTCs. IIUC, it's something like "GHC current limits quantified constraints to exclude type families".

Reply:
Another idea: are there any additional parameters we could anticipate adding and just add them now?

Reply:
There will be some additions: we don't have: `TArray`, `TChan` and `TSem` from [stm](https://hackage.haskell.org/package/stm), but they need more work to be introduced (so not in this pr).